components/script/dom/console.rs

```diff
@@ -65,13 +73,80 @@ fn console_message(global: &GlobalScope, message: DOMString, level: LogLevel) {
     })
 }
 
+//generate matrix of table elements
+#[allow(unsafe_code)]
+unsafe fn generate_table(
+    cx: &SafeJSContext,
+    tabular_data: &HandleValue,
+    _properties: &Option<Vec<DOMString>>,
+) -> Option<Vec<Vec<DOMString>>> {
+    let cx_ptr = *(cx.deref()) as *mut JSContext;
+
+    if !tabular_data.get().is_object() {
+        return None;
+    }
+
+    if is_array_like(cx_ptr, *tabular_data) {
+        let vec: Vec<DOMString> =
+            Vec::<DOMString>::from_jsval(cx_ptr, *tabular_data, StringificationBehavior::Empty)
+                .expect("Error when converting Array object to Vec<DOMString>")
+                .get_success_value()
+                .unwrap_or(&Vec::new())
+                .to_vec();
+
+        let mut res: Vec<Vec<DOMString>> = Vec::new();
+        res.push(vec!["(index)".into(), "Values".into()]);
+        for (index, word) in vec.into_iter().enumerate() {
+            res.push(vec![index.to_string().into(), word]);
+        }
+
+        return Some(res);
+    }
+
+    None
+}
+
+//transform matrix of table elements into DOMString for logging
+fn generate_table_as_string(table: &[Vec<DOMString>]) -> DOMString {
+    let mut res: DOMString = DOMString::new();
+    for row in table.iter() {
+        for col in row.iter() {
+            res.push_str(col as &str);
+            res.push_str("\t\t");
```

Comment:
If we implement this, it deserves an implementation that knows how to properly align the columns of the table. Two tabs are pretty brittle, so maybe investigate if there is a more resilient way to do this based on the actual strings that we are processing.

Reply:
That's a good idea. I'll see if there's a good way to align the columns.

Reply:
I've implemented a simple way to align the columns by setting each column's width to be the maximum length of all of its rows, plus some extra space.